glean-core/python/glean/_loader.py

```diff
@@ -0,0 +1,146 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+"""
+Utilities for loading metrics.yaml and pings.yaml files and creating a tree of
+metric types.
+"""
+
+
+from pathlib import Path
+from typing import Any, List, Union
+
+
+from glean_parser.parser import parse_objects  # type: ignore
+import glean_parser.metrics  # type: ignore
+
+
+from .metrics import CounterMetricType, PingType
+
+
+# A mapping from the name of the metric type as it appears in the metrics.yaml
+# to the Python class for that metric type.
+_TYPE_MAPPING = {"counter": CounterMetricType, "ping": PingType}
+
+
+# The arguments that should be passed to the constructor for the metric types.
+_ARGS = [
+    "allowed_extra_keys",
+    "bucket_count",
+    "category",
+    "disabled",
+    "histogram_type",
+    "include_client_id",
+    "lifetime",
+    "memory_unit",
+    "name",
+    "range_max",
+    "range_min",
+    "send_in_pings",
+    "time_unit",
+]
+
+
+class UnsupportedMetricType:
```

Comment:
Are we planning on removing this once we reach feature parity? If so, can you please file a bug about removing it and add the bug number as a TODO comment here?

Reply:
We'll probably never remove this -- we should have some way to handle metric types the Python bindings don't support (maybe "CustomDistributions") without needing to throw out the entire metrics.yaml file.

Reply:
I see, ok!